### PR TITLE
CLOUDP-271289: updated golangci-lint to 1.60.3 and fixed new lintintg errors

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
## Proposed changes
updated golangci-lint to 1.60.3 and fixed new lintintg errors

_Jira ticket:_ CLOUDP-271289

### CODE REVIEW

File: internal/cli/watch_opts.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ type WatchOpts struct {
 	s              *spinner.Spinner
 	EnableWatch    bool
 	DefaultWait    time.Duration
-	Timeout        uint
```
Comment (gssbzn) [inline internal/cli/watch_opts.go:32]: [q] is `uint` discouraged now?

Comment (jeroenvervaeke) [reply internal/cli/watch_opts.go:32]: The issue is that the places where it's used (like [here](https://github.com/mongodb/mongodb-atlas-cli/blob/46ea2ba44076b660fd3c8eb2e2568a75f10f0958/internal/cli/clusters/create.go#L125)), we always convert from `uint` to `time.Duration`, which is `int64`. Linting complains about possible overflows.

Example usage
```go
watcher.Timeout = time.Duration(opts.Timeout)
```

**Review by fmenezes — APPROVED** [review]
LGTM

**Review by gssbzn — APPROVED** [review]

File: .golangci.yml [hunk 2]
```diff
@@ -109,7 +109,7 @@ linters:
     - errcheck # Errcheck is a program for checking for unchecked errors in go programs. These unchecked errors can be critical bugs in some cases [fast: false, auto-fix: false]
     - errorlint # Errorlint is a linter that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13.
     - exhaustive # check exhaustiveness of enum switch statements [fast: false, auto-fix: false]
-    - exportloopref # checks for pointers to enclosing loop variables [fast: false, auto-fix: false]
+    - copyloopvar # copyloopvar is a linter detects places where loop variables are copied. Replaces exportloopref since Go1.22
```
Comment (davidhou17) [inline .golangci.yml:112]: change to "copyloopvar is a linter **that** detects places where ..."

**Review by davidhou17 — COMMENTED** [review]
LGTM % 1 fix

File: .golangci.yml (hunk 2 above)
Comment (davidhou17) [reply .golangci.yml:112]: Ah, I see this has already been merged. Please hold off on merging next time until the Cloud docs review window has closed, thanks

Comment (jeroenvervaeke) [reply .golangci.yml:112]: My bad, didn't see that there were docs changes.
I didn't write this text; it's a copy-paste from the [documentation](https://golangci-lint.run/usage/linters/#copyloopvar).

**Pull request merged**